tests/poll.rs

```diff
@@ -220,6 +329,56 @@ pub fn double_register() {
         .is_err());
 }
 
+// Testing on Windows only where this must fail by design, on other platforms
+// behavior is undefined, for instance on kqueue platforms reregistering w/o
+// registering works.
+#[test]
+#[cfg(target_os = "windows")]
+fn reregister_without_register() {
+    /*
+    This test checks the following reregister contraint:
+    The event source must have previously been registered with this instance
+    of `Poll` otherwise the call to `reregister` will return with an error.
+    */
+
+    let poll = Poll::new().expect("unable to create Poll instance");
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+
+    assert_eq!(
+        poll.registry()
+            .reregister(&mut listener, ID1, Interest::READABLE)
+            .unwrap_err()
+            .kind(),
+        io::ErrorKind::NotFound
+    );
+}
+
+// Testing on Windows only where this must fail by design, on other platforms
+// behavior is undefined, for instance on kqueue platforms deregistering w/o
+// registering works.
+#[test]
+#[cfg(target_os = "windows")]
+fn deregister_without_register() {
+    /*
+    This test checks the following reregister constraint applied to deregister:
```

Comment:
Wording: "This test checks the following deregister constraint:"

I wouldn't rely on the previous test to describe this

Reply:
redone.